app/controllers/messages_controller.rb

```diff
@@ -5,16 +5,22 @@ class MessagesController < ApplicationController
 
   def index
     blocked_ids = Block.where(blocking_user: current_user).pluck(:blocked_user_id)
+
+    includes = []
     if params[:view] == 'outbox'
       @page_title = 'Outbox'
       from_table = current_user.sent_messages.where(visible_outbox: true).ordered_by_thread.select('distinct on (thread_id) messages.*')
       from_table = from_table.where.not(recipient_id: blocked_ids).joins(:recipient).where.not(users: { deleted: true })
+      includes << :recipient
     else
       @page_title = 'Inbox'
       from_table = current_user.messages.where(visible_inbox: true).ordered_by_thread.select('distinct on (thread_id) messages.*')
       from_table = from_table.where.not(sender_id: blocked_ids).left_outer_joins(:sender).where('users.deleted IS NULL OR users.deleted = false')
+      includes << :sender
     end
-    @messages = Message.from(from_table).select('*').order('subquery.id desc').paginate(page: page)
+    @messages = Message.from(from_table, "messages").joins(:first_thread)
+      .select('*', 'first_threads_messages.subject as thread_subject')
+      .includes(*includes).order('messages.id desc').paginate(page: page)
```

Comment:
The includes here are for displaying the other participant in messages on message list views, which is currently a n+1

Reply:
:deer:

Reply:
why'd we change the order statement here?

Reply:
It should be doing the same thing - we just renamed the name Rails will give to the subquery within the `.from` call. I think it was necessary so it knew which field to join on when I added `.joins(:first_thread)`.